test/buildSettingsInheritanceCheck/diamond/p1/dub.sdl

```diff
@@ -0,0 +1,5 @@
+name "p1"
+versions "B"
```

Comment:
Better use a matching version `P1`, so it's easier to understand where the version comes from.
How about using `{grand,}{parent,mother,father}`, `daughter`, and/or `son` as terminology. Might simplify understanding.